torch/_dynamo/guards.py

```diff
@@ -414,9 +418,13 @@ def TENSOR_MATCH(self, guard: Guard):
             self.ID_MATCH(guard)
         else:
             value = self.get(guard.name)
-            self.tensor_check_names.append(self.arg_ref(guard))
+            tensor_name = self.arg_ref(guard)
```

Comment:
noob Q: what does arg_ref do? I read the implementation but I cannot tell what the returned name is supposed to denote.

Reply:
Discussed offline

Reply:
for others: the name is the name torchdynamo uses to refer to the variable in the frame when the guards are being evaluated. But there's also some extra wibbling around base name that neither of us know how it works